frame/core-fellowship/src/benchmarking.rs

```diff
@@ -0,0 +1,200 @@
+		assert!(MemberEvidence::<T, I>::contains_key(who));
+	}
+
+	fn make_member<T: Config<I>, I: 'static>(rank: u16) -> T::AccountId {
+		let member = account("member", 0, SEED);
+		T::Members::induct(&member).unwrap();
+		for _ in 0..rank {
+			T::Members::promote(&member).unwrap();
+		}
+		CoreFellowship::<T, I>::import(RawOrigin::Signed(member.clone()).into()).unwrap();
+		member
+	}
+
+	#[benchmark]
+	fn set_params() {
+		let params = ParamsType {
+			active_salary: [100u32.into(); 9],
+			passive_salary: [10u32.into(); 9],
+			demotion_period: [100u32.into(); 9],
+			min_promotion_period: [100u32.into(); 9],
+			offboard_timeout: 1u32.into(),
+		};
+
+		#[extrinsic_call]
+		_(RawOrigin::Root, Box::new(params.clone()));
+
+		assert_eq!(Params::<T, I>::get(), params);
+	}
+
+	#[benchmark]
+	fn bump_offboard() {
+		let member = make_member::<T, I>(0);
+
+		// Set it to the max value to ensure that any possible auto-demotion period has passed.
+		frame_system::Pallet::<T>::set_block_number(T::BlockNumber::max_value());
+		ensure_evidence::<T, I>(&member);
+		assert!(Member::<T, I>::contains_key(&member));
+
+		#[extrinsic_call]
+		CoreFellowship::<T, I>::bump(RawOrigin::Signed(member.clone()), member.clone());
+
+		assert!(!Member::<T, I>::contains_key(&member));
+		assert!(!MemberEvidence::<T, I>::contains_key(&member));
+	}
+
+	#[benchmark]
+	fn bump_demote() {
+		let member = make_member::<T, I>(2);
+
+		// Set it to the max value to ensure that any possible auto-demotion period has passed.
+		frame_system::Pallet::<T>::set_block_number(T::BlockNumber::max_value());
+		ensure_evidence::<T, I>(&member);
+		assert!(Member::<T, I>::contains_key(&member));
+		assert_eq!(T::Members::rank_of(&member), Some(2));
+
+		#[extrinsic_call]
+		CoreFellowship::<T, I>::bump(RawOrigin::Signed(member.clone()), member.clone());
+
+		assert!(Member::<T, I>::contains_key(&member));
+		assert_eq!(T::Members::rank_of(&member), Some(1));
+		assert!(!MemberEvidence::<T, I>::contains_key(&member));
+	}
+
+	#[benchmark]
+	fn set_active() {
+		let member = make_member::<T, I>(1);
+		assert!(Member::<T, I>::get(&member).unwrap().is_active);
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(member.clone()), false);
+
+		assert!(!Member::<T, I>::get(&member).unwrap().is_active);
+	}
+
+	#[benchmark]
+	fn induct() {
+		let candidate: T::AccountId = account("candidate", 0, SEED);
+
+		#[extrinsic_call]
+		_(RawOrigin::Root, candidate.clone());
+
+		assert_eq!(T::Members::rank_of(&candidate), Some(0));
+		assert!(Member::<T, I>::contains_key(&candidate));
+	}
+
+	#[benchmark]
+	fn promote() {
+		let member = make_member::<T, I>(1);
+		ensure_evidence::<T, I>(&member);
+
+		#[extrinsic_call]
+		_(RawOrigin::Root, member.clone(), 2u8.into());
+
+		assert_eq!(T::Members::rank_of(&member), Some(2));
+		assert!(!MemberEvidence::<T, I>::contains_key(&member));
+	}
+
+	#[benchmark]
+	fn offboard() {
+		let member = make_member::<T, I>(0);
+		T::Members::demote(&member).unwrap();
```

Comment:
Just FYI: the V2 syntax supports `?` if you change the return type of the benchmark to `Result<(), BenchmarkError>`.